js/dactylo.js

```diff
@@ -87,7 +87,6 @@ window.addEventListener('DOMContentLoaded', () => {
     const lessonLetters = rawLetters.concat(!has1dk ? [] :
       rawLetters.flatMap(letter => (letter in odk ? [odk[letter]] : [])));
```

Comment:
`array.filter().map().reduce()` : 80% of the time, it works all the time. Jokes aside, functional programming is usually not about writing a single expression that does everything in one go, but feeding your data through a pipeline, especially with containers.

The procedural way of building `lessonLetters` would be "if you have a value associated with that letter in `odk`, add the value to the array", whereas with the pipeline analogy you would say "take the array, discard any letter that doesn’t have a value in `odk`, *then* get the value for all of the letters left". "then" is the important word in this analogy, it means you’re supposed to add another function to your pipeline, you’re doing another action.

Pipeline analogies aside, this whole `!has1dk ? [] : <bunch of stuff>` directly as a function argument is really confusing in my eyes, and deserves to be bound to a variable and given a name. Here’s how I would do it : 

```js
    const deadKeyLetters = !has1dk ? [] :
      rawLetters
        .filter(letter => letter in odk)
        .map(letter => odk[letter]);

    const lessonLetters = rawLetters.concat(deadKeyLetters);
```

Reply:
OMG you’re totally right. I used `.flatMap` a couple lines before for a good reason, and then used it here instead of filter, which was totally stupid. Thanks for pointing it out.